### DISCUSSION TITLE
tox evaluates tests only for py38, ignores py36/py37

### DISCUSSION THREAD

**Discussion by hbaniecki** [ymyzk/tox-gh-actions#159] [Q&A] [answered]
Hi,
Your package is great! It allowed me to easily plug-in `gh-actions` for python and `unittest`.
The problem that occurred for me is that only tests for `py38` are evaluated. `py36/py37` are successful but actually nothing happens after running the `tox` command.
e.g.:
- [ubuntu 3.6/Test with tox](https://github.com/ModelOriented/DALEX/runs/676169198?check_suite_focus=true) 1 sec with success
- [ubuntu 3.8/Test with tox](https://github.com/ModelOriented/DALEX/runs/676169216?check_suite_focus=true) indeed did the test

What did I miss? [Python-check.yml](https://github.com/ModelOriented/DALEX/blob/python-GA/.github/workflows/Python-check.yaml) &emsp; [tox.ini](https://github.com/ModelOriented/DALEX/blob/python-GA/tox.ini)

**Comment by ymyzk** [comment, marked as answer]
Hi @hbaniecki, I think this is expected behavior. Here's how tox-gh-actions understand your configuration.

`tox.ini` defines the following 5 test environments:
- py36
- py38-linux
- py38-macos
- py38-windows
- flake8
You can check this by running `tox -a`

`Python-check.yml` generates 9 tests configurations:
- Python {3.6,3.7,3.8} with PLATFORM {ubuntu-latest,macos-latest,windows-latest} 

py36 test environment doesn't satisfy requirements as it doesn't have a platform factor like linux.

---

To fix your problem, I'd suggest to update `tox.ini` to
```
envlist = py{36,37,38}-{linux,macos,windows}
```

Furthermore, if you don't need to write platform specific configurations in `tox.ini`. You can drop configuration for platform.
```
envlist = py36, py37, py38
```
```
    - name: Test with tox
      run: tox
```

**Comment by hbaniecki** [comment]
Thanks!
